### PR TITLE
Added support to audit the status of a service-linked role deletion.

### PR DESCRIPTION
**Pull request by Nirbhay1997**
Signed-off-by: nirbhay1997 <nkumar@progress.com>

### Description

Added support to audit the status of a service-linked role deletion.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [x] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_iam_service_linked_role_deletion_status.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    return nil unless exists?
+    @res[:status]
+  end
+
+  def exists?
+    !@res.nil? && !@res.empty?
+  end
+
+  def to_s
+    "deletion_task_id: #{@display_name}"
```
Comment (soumyo13) [inline libraries/aws_iam_service_linked_role_deletion_status.rb:38]:
```suggestion
    "Deletion Task ID: #{@display_name}"
```

File: test/unit/resources/aws_iam_service_linked_role_deletion_status_test.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    data = {}
+    data[:method] = :get_service_linked_role_deletion_status
+    mock_parameter = {}
+    mock_parameter[:status] = 'SUCCEEDED'
+    data[:data] = mock_parameter
+    data[:client] = Aws::IAM::Client
+    @response = AWSIAMServiceLinkedRoleDeletionStatus.new(deletion_task_id: 'test1', client_args: { stub_responses: true }, stub_data: [data])
+  end
+
+  def test_parameter_group_exists
```
Comment (soumyo13) [inline test/unit/resources/aws_iam_service_linked_role_deletion_status_test.rb:32]: Change the method name.

File: docs/resources/aws_iam_service_linked_role_deletion_status.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+Ensure that a Service Linked Role deletion task exists.
+
+    describe aws_iam_service_linked_role_deletion_status(deletion_task_id: 'deletion_task_id') do
+      it { should exist }
+    end
+
+## Parameters
+
+`deletion_task_id` _(required)_
+
```
Comment (soumyo13) [inline docs/resources/aws_iam_service_linked_role_deletion_status.md:21]: Add what is deletion_task_id!

File: docs/resources/aws_iam_service_linked_role_deletion_status.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+| --- | --- |
+| status | The name identifying the instance profile. |
+
+
+
+## Examples
+
+### Ensure a Service Linked Role is deleted.
+
+    describe aws_iam_service_linked_role_deletion_status(deletion_task_id: 'deletion_task_id') do
```
Comment (soumyo13) [inline docs/resources/aws_iam_service_linked_role_deletion_status.md:36]: Change all the deletion_task_id to some meaningful values.

**Review by soumyo13 — CHANGES_REQUESTED** [review]
LGTM. Please add the tests.

**Comment by soumyo13** [comment]
Please change the description of the PR.

**Review by soumyo13 — APPROVED** [review]
LGTM

**Review by IanMadd — APPROVED** [review]

**Pull request merged**